src/util/syscall_sandbox.cpp

```diff
@@ -0,0 +1,882 @@
+        // abilities can never be regained.
+        //
+        // SyscallSandboxPolicy::INITIALIZATION must thus be a superset of all
+        // other policies.
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::INITIALIZATION_DNS_SEED: // Thread: dnsseed
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::INITIALIZATION_LOAD_BLOCKS: // Thread: loadblk
+        seccomp_policy_builder.AllowFileSystem();
+        break;
+    case SyscallSandboxPolicy::INITIALIZATION_MAP_PORT: // Thread: mapport
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::MESSAGE_HANDLER: // Thread: msghand
+        seccomp_policy_builder.AllowFileSystem();
+        break;
+    case SyscallSandboxPolicy::NET: // Thread: net
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::NET_ADD_CONNECTION: // Thread: addcon
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::NET_HTTP_SERVER: // Thread: http
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::NET_HTTP_SERVER_WORKER: // Thread: httpworker.<N>
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::NET_OPEN_CONNECTION: // Thread: opencon
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::SCHEDULER: // Thread: scheduler
+        seccomp_policy_builder.AllowFileSystem();
+        break;
+    case SyscallSandboxPolicy::TOR_CONTROL: // Thread: torcontrol
+        seccomp_policy_builder.AllowFileSystem();
+        seccomp_policy_builder.AllowNetwork();
+        break;
+    case SyscallSandboxPolicy::TX_INDEX: // Thread: txindex
+        seccomp_policy_builder.AllowFileSystem();
+        break;
+    case SyscallSandboxPolicy::VALIDATION_SCRIPT_CHECK: // Thread: scriptch.<N>
+        break;
+    case SyscallSandboxPolicy::SHUTOFF: // Thread: main thread (state: shutoff)
+        seccomp_policy_builder.AllowFileSystem();
+        break;
+    }
+
+    const SyscallSandboxAction default_action = g_syscall_sandbox_log_violation_before_terminating ? SyscallSandboxAction::INVOKE_SIGNAL_HANDLER : SyscallSandboxAction::KILL_PROCESS;
+    std::vector<sock_filter> filter = seccomp_policy_builder.BuildFilter(default_action);
+    const sock_fprog prog = {
+        .len = static_cast<uint16_t>(filter.size()),
+        .filter = filter.data(),
+    };
```

Comment:
https://en.cppreference.com/w/cpp/language/aggregate_initialization

Neither I, nor CI could find a compiler that rejects this C++20 code, so does that mean we are allowed to use it in new code now?

Reply:
Looks like this is supported since clang 3.2 and gcc 4.7, but maybe not msvc?